types/artifacts.d.ts

```diff
@@ -743,6 +741,56 @@ declare global {
         /** Column number in the script (0-based). */
         columnNumber: number;
       }
+
+      /** Describes a generic console message. */
+      interface BaseConsoleMessage {
+        /**
+         * What emitted the console message (console API, violation log, etc.).
+         * See subtypes for possible values.
+         */
+        source: string,
+        /** The log level: info, warning, etc. See subtypes for possible values. */
+        level: string,
+        /**
+         * The text printed to the console, as shown on the browser. Note objects printed to the
+         * console will be stringified if primitive or JSON-serializable. Otherwise it will be
+         * a String representation of the object.
+         */
+        text: string;
+        /** Time of the console log in milliseconds since epoch. */
+        timestamp: number;
+        /** The stack trace of the log/exception, if known. */
+        stackTrace?: Crdp.Runtime.StackTrace;
+        /** The URL of the log/exception, if known. */
+        url?: string;
+        /** Line number in the script (0-based), if known. */
+        lineNumber?: number;
+        /** Column number in the script (0-based), if known. */
+        columnNumber?: number;
+      }
+
+      /** Describes a console message logged by a script using the console API. */
+      interface ConsoleAPICall extends BaseConsoleMessage {
+        /** The console API invoked. Only the following console API calls are gathered. */
+        source: 'console.warn' | 'console.error';
+        /** Corresponds to the API call. */
+        level: 'warning' | 'error';
+      }
+
+      interface ConsoleException extends BaseConsoleMessage {
+        source: 'exception';
+        level: 'error';
+      }
+
+      /** Describes a console message logged by the browser for violation reporting. */
+      interface ConsoleLogEntry extends BaseConsoleMessage {
```

Comment:
it's still a bit confusing to me that `ConsoleLogEntry` is the set of violations rather than "entries that came from `console.log`" :)

WDYT about `ConsoleProtocolEntry` or anything that helps signify the `Log` part means the CDP `Log` domain and not `ConsoleLog`

Reply:
Went with `ConsoleBrowserReport`. `Protocol` sounds like it's internal to me but it's certainly visible on the log

Reply:
I dig it 🕳️  👍